typescript/src/main.ts

```diff
@@ -178,35 +178,72 @@ function reset() {
   calculate();
 }
 
+function cleanData(data: string) {
+  data = data.replace(/(\r\n|\n|\r)/gm, "").replace(/(\s\s)/gm, " ");
+  data = data.replace(/"/g, '""');
+  data = data.replace(/\u2206/g, "Delta ");
+  return data;
+}
+function convertToPlainText(html: string) {
+  // Create a new div element
+  const tempDivElement = document.createElement("div");
+  // Set the HTML content with the given value
+  tempDivElement.innerHTML = html;
+  // Retrieve the text property of the element
+  return tempDivElement.textContent || tempDivElement.innerText || "";
+}
+
 // Quick and simple export target #tableId into a csv
 function downloadTableAsCsv(tableId: string, separator = ",") {
-  // Select rows from tableId
-  const rows = document.querySelectorAll("table#" + tableId + " tr");
+  const table = <HTMLTableElement>document.getElementById(tableId);
+  const rows = table.rows;
   // Construct csv
   const csv = [];
   for (let i = 0; i < rows.length; i++) {
     const row = [];
-    const cols = <NodeListOf<HTMLTableCellElement>>(
-      rows[i].querySelectorAll("td, th")
-    );
+    const cols = rows[i].cells;
     for (let j = 0; j < cols.length; j++) {
       // Clean innertext to remove multiple spaces and jumpline (break csv)
       let data;
-      const inputs = cols[j].getElementsByTagName("input");
+
+      const unfilteredInputs = cols[j].getElementsByTagName("input");
+      const inputs = [];
+      if (unfilteredInputs.length > 0) {
+        for (let l = 0; l < unfilteredInputs.length; l++) {
+          const input = unfilteredInputs[l];
+          if (input.type == "number") {
+            inputs.push(input);
+          }
+        }
+      }
+
       if (inputs.length > 0) {
         data = inputs[0].value;
       } else {
         data = cols[j].innerText;
       }
       // Remove line breaks and escape double-quote with double-double-quote
-      data = data.replace(/(\r\n|\n|\r)/gm, "").replace(/(\s\s)/gm, " ");
-      data = data.replace(/"/g, '""');
-      data = data.replace(/\u2206/g, "Delta ");
+      data = cleanData(data);
       // Push escaped string
       row.push('"' + data + '"');
     }
     csv.push(row.join(separator));
   }
+  const explanation = [];
+  const headings = rows[0].cells;
+  for (let k = 0; k < headings.length; k++) {
+    const spans = headings[k].getElementsByTagName("span");
+    if (spans.length > 0) {
+      let span = spans[0].innerHTML;
```

Comment:
Just a small thing but it might be better to name this e.g. `spanText` or `detailText` to show that it's a string rather than an HTML Element. Ditto with header.

Reply:
Pardon my lack of slang but does ditto mean its alright or change that as well?

Reply:
Sorry! ditto means the same applies to, so please change header too.